bulkio: Support running BACKUP, IMPORT and RESTORE under transaction. …

miretskiy opened the pull request:
Informs #47539

Allow execution of BACKUP, IMPORT, and RESTORE statements
under explicit transaction, provided the "DETACHED" option is specified.

Release notes (improvement): BACKUP, IMPORT, and RESTORE can run
under explicit transaction provided DETACHED option is specified.

Commit:
sql: Add DETACHED option to IMPORT, BACKUP and RESTORE

Add DETACHED option to the IMPORT, BACKUP, and RESTORE statements.

For example:
  IMPORT INTO t CSV DATA (...) DETACHED
  BACKUP DATABASE foo TO ... DETACHED

When specified, the statement will run in "detached" mode.
That is, the caller will get back job ID, but the caller will
not be blocked for the duration of the operation.

The result of the statement can be examined via SHOW JOBS.

Release notes (sql change): Add DETACHED option to long running
queries that execute as jobs: IMPORT, BACKUP, RESTORE.
This option, when specified for these commands, does not block
the caller.  The job runs in "detached" mode, and the caller can
examine the result of the command using SHOW JOBS

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49172)
<!-- Reviewable:end -->

knz commented:
before I review this I'd like an answer on this thread: https://github.com/cockroachdb/cockroach/issues/47539#issuecomment-632034162

miretskiy commented:
@knz commented on the issue; lmk what you think.

Commit:
bulkio: Support running BACKUP, IMPORT and RESTORE under transaction.

Allow execution of BACKUP, IMPORT, and RESTORE statements
under explicit transaction, provided the "DETACHED" option is specified.

Release notes (improvement): BACKUP, IMPORT, and RESTORE can run
under explicit transaction provided DETACHED option is specified.
